### PR TITLE
Add INSTALL.md

### PR DESCRIPTION
**Pull request by jefft0**
Add INSTALL.md with instructions for macOS and Ubuntu.

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    export PATH="$PATH:$ANDROID_HOME/platform-tools"
+    export JAVA_HOME="/usr/local/android-studio/jre"
+    export PATH="$PATH:$JAVA_HOME/bin"
+
+In a terminal, enter:
+
+    git clone https://github.com/ipfs-shipyard/gomobile-ipfs
+    cd gomobile-ipfs/packages
+    make build_core.android
+    make test_bridge.android
```
Comment (aeddi) [inline INSTALL.md:109]: It seems to me that the rule `make test_bridge.android` will launch `make build_core.android` by itself.

I think you could also explain to people that this part tells them how to run the tests on the bridge; someone might only want to build the core or the Example app and would not be interested in this step at all.

**Review by aeddi — APPROVED** [review]

File: INSTALL.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
+Build gomobile-ipfs
+===================
+
+These are instructions to build gomobile-ipfs.
+
+Prerequisites
+=============
+* Required: Git
+* Required: Go version 1.17
+* Required: Python modules pystache and yaml
```
[the remark this answers is not in the document]
Comment (jefft0) [reply INSTALL.md:10]: This works for me on macOS 11 and 12, but not on macOS 10.15 . Should we try to make it work on macOS 10.15 ?

File: INSTALL.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+To install Go 1.17, follow instructions at https://go.dev/dl .
+
+To install Python module pystache, in a terminal enter:
+
+    sudo python3 -m pip install pystache
+
+To install Python module yaml, in a terminal enter:
+
+    sudo python3 -m pip install pyyaml
```
[the remark this answers is not in the document]
Comment (jefft0) [reply INSTALL.md:29]: But not on macOS 10.15. See my comment above.

File: INSTALL.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+These are instructions to build gomobile-ipfs.
+
+Prerequisites
+=============
+* Required: Git
+* Required: Go version 1.17
+* Required: Python modules pystache and yaml
+* Required to build for Android: Android Studio
+* Required to build for iOS: cocoapods
```
[the remark this answers is not in the document]
Comment (jefft0) [reply INSTALL.md:12]: OK. Will do.

File: INSTALL.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+Following are the steps to install each prerequisite (if it's needed for your build target).
+
+## macOS 10.15, macOS 11 and macOS 12
+Install Xcode.
+
+(Git is already installed.)
+
+To install Go 1.17, follow instructions at https://go.dev/dl .
```
[the remark this answers is not in the document]
Comment (jefft0) [reply INSTALL.md:21]: OK. Will do.

File: INSTALL.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+To launch Android Studio, in a terminal enter:
+
+    /usr/local/android-studio/bin/studio.sh &
+
+Build for Android
+=================
+* Launch Android Studio and accept the default startup options. Create a new default project (so that we get the main screen).
+* On the Tools menu, open the SDK Manager.
+* In the "SDK Platforms" tab, check "Android 10.0 (Q)".
```
[the remark this answers is not in the document]
Comment (jefft0) [reply INSTALL.md:73]: Gomobile-IPFS does not build with Android 11. That's why I said Android 10. Should I say "mandatory Android 10 or earlier"?

File: INSTALL.md (hunk 1 above)
Comment (jefft0) [reply INSTALL.md:109]: You are right. I will update it.

File: INSTALL.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+In a terminal, enter for following with the correct <version> to install in /usr/local .
+
+    sudo tar -C /usr/local -xzf go<version>.linux-amd64.tar.gz
+
+To install Python module pystache, in a terminal enter:
+
+    sudo apt install python3-pip
+    sudo python3 -m pip install pystache
+
+(Python module yaml is already installed.)
```
[the remark this answers is not in the document]
Comment (jefft0) [reply INSTALL.md:58]: You are right. I will change it to only require pip3.

File: INSTALL.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jefft0) [reply INSTALL.md:10]: Building on macOS 10.15 gives the following error. (pip3 is already installed, there is a different problem):
maven_format_core.py && touch /Users/remap/gomobile-ipfs/packages/build/android/maven/ipfs/gomobile/core/0.0.42-dev
Traceback (most recent call last):
  File "/Users/remap/gomobile-ipfs/packages/utils/maven_format/maven_format_core.py", line 4, in <module>
    import pystache
ModuleNotFoundError: No module named 'pystache'

File: INSTALL.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+## Ubuntu 18.04, 20.04 and 22.04
+
+To install Git, in a terminal, enter:
+
+    sudo apt install git
+
+To install go 1.17, download go<version>.linux-amd64.tar.gz from https://go.dev/dl.
+In a terminal, enter for following with the correct <version> to install in /usr/local .
+
+    sudo tar -C /usr/local -xzf go<version>.linux-amd64.tar.gz
```
[the remark this answers is not in the document]
Comment (jefft0) [reply INSTALL.md]: I changed it to "Install go 1.17 with a package manager, or follow instructions at https://go.dev/dl"

File: INSTALL.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jefft0) [reply INSTALL.md]: The problem on macOS 10.15 was an outdated pip3. I changed to instructions only on macOS 10.15 to upgrade pip3.

File: INSTALL.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jefft0) [reply INSTALL.md]: ... As mentioned above, I changed to instructions only on macOS 10.5 to upgrade pip3.

File: INSTALL.md (hunk 1 above)
Comment (jefft0) [reply INSTALL.md]: I changed it to say look at `make help` and that other targets build the core.

File: INSTALL.md [hunk 9]
```diff
@@ -0,0 +1,124 @@
+Build gomobile-ipfs
+===================
+
+These are instructions to build gomobile-ipfs.
+
+Prerequisites
+=============
+* Required: Git
```
[the remark this answers is not in the document]
Comment (jefft0) [reply INSTALL.md]: I tested again and it seems that Go works on Ubuntu without installing Git. So I removed the requirement as you said.

File: INSTALL.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (aeddi) [reply INSTALL.md:14]: We should change the behaviour of this script maven_format_core.py and auto-detect the path of the Android SDK, so the user would be able to use the Android version he wants.
But for now, we can keep this line as it is 👍

File: INSTALL.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (aeddi) [reply INSTALL.md:72]: We should change the behaviour of this script maven_format_core.py and auto-detect the path of the Android SDK, so the user would be able to use the Android version he wants.
But for now, we can keep this line as it is 👍

File: INSTALL.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (jefft0) [reply INSTALL.md:15]: I changed from Xcode to install the Command Line Developer Tools using `xcode-select --install` and Software Update. Tested on fresh installations of macOS 10.5, macOS 11 and macOS 12.

**Pull request merged**